ops/charm.py

```diff
@@ -831,6 +831,60 @@ class PebbleCustomNoticeEvent(PebbleNoticeEvent):
     """Event triggered when a Pebble notice of type "custom" is created or repeats."""
 
 
+class PebbleCheckEvent(WorkloadEvent):
+    """Base class for Pebble check events."""
+
+    info: model.LazyCheckInfo
+    """Provide access to the check's current state."""
+
+    def __init__(
+        self,
+        handle: Handle,
+        workload: model.Container,
+        check_name: str,
+    ):
+        super().__init__(handle, workload)
+        self.info = model.LazyCheckInfo(workload, check_name)
+
+    def snapshot(self) -> Dict[str, Any]:
+        """Used by the framework to serialize the event to disk.
+
+        Not meant to be called by charm code.
+        """
+        d = super().snapshot()
+        d['check_name'] = self.info.name
+        return d
```

Comment:
Super-minor: why `d`? because it's a dict? delta? data?

Reply:
How about:

```suggestion
        return {**super().snapshot(), "check_name": self.info.name}
```

Reply:
I'm not a fan of the `d` either, but it's consistent with all of the others. @benhoyt what's your take on this? A separate PR to adjust them all? Have this one be inconsistent? Change them all in this PR as a drive-by? Leave it alone for now?

Reply:
I like a separate PR to adjust them all would be fine. Probably `dct` or `snapshot`?

Reply:
@dimaqq would you like to open one for that? I can do it if not.

Reply:
My vote is for an inline dictionary merge, as it obviates local variable entirely, while is just as readable.